src/v2/Apps/Order/Routes/Payment/index.tsx

```diff
@@ -149,6 +151,20 @@ export const PaymentRoute: FC<Props> = props => {
     })
   }
 
+  const setupIntentId = getClientParam("setup_intent")
+  const isSettingPayment =
+    setupIntentId &&
+    getClientParam("setup_intent_client_secret") &&
+    getClientParam("redirect_status") === "succeeded"
```

Comment:
After redirecting back from Stripe, we get 3 query params. We'll use them as an indication to set payment behind the scene. We may consider making it stricter by checking order state or something.